sdk/nodejs/cmd/run-policy-pack/index.ts

```diff
@@ -104,9 +104,10 @@ function main(args: string[]): void {
 
     // Ensure that our v8 hooks have been initialized.  Then actually load and run the user program.
     v8Hooks.isInitializedAsync().then(() => {
+        programRunning = true;
         const promise: Promise<void> = require("./run").run({
             argv,
-            programStarted: () => (programRunning = true),
+            programStarted: () => programRunning,
```

Comment:
is this a correct change? Program started seems to be a hook to set programRunning to true at a later time where before programRunning currently is able to be false before programStarted is called to set it to true.

Reply:
Good catch!
Previously, programRunning would be true only if programStarted had been called.
This assignment should happen inside the lambda.

Reply:
Actually, _I think_ this lint caught an actual bug here, but I'm going to double-check. Here's my thinking.

This is my understanding of how the program is meant to function:

1. We load the `./run` module and call the `run` function inside of that module.
2. That function loads the user program and executes it.
3. Once the `run` function completes, we set `programRunning = false` on line 123. I would expect `programRunning` to always be false after the program has terminated.

The way this code _was_ written, you could call `programStarted()` after the program had already exited and you'd set `programRunning` to true again.  I don't think it's possible to have the program halt and then start back up again in the same invocation of the Language runtime (i.e. this `index.ts` file).

Suppose after the program has completed, another thread calls `programStarted()` to determine whether or not the program has started. `programStarted()` will return true and also update `programRunning` to true, without running the program again.


However, I think Kyle's question is still largely unanswered:

> Can you explain why programStarted() exists as a parameter before merging this? 🙂

Is `programStarted()` a callback intended to signal that `programRunning` should be set to true? If so, why does it return a boolean? It should be called `startProgram`. In this case, the change I made is incorrect for the reasons you've both noted. 😸 

Or is `programStarted()` intended to assay whether the program _was_ started? In this case, the lint has identified a bug. 🐛 The author probably intended to use `===` or `==` instead of `=`.

Reply:
So _I think_ that means you're right!
Man, assignment expressions **are confusing**!

Reply:
I think they meant to do something like

```typescript
() => {
    programRunning = true;
}
```
and the return value is not relevant, but you can omit braces to make it a one-liner.